python/integrationtest/log_file_checks.py

```diff
@@ -1,31 +1,74 @@
 from glob import glob
 import re
 
-def log_has_no_errors(log_file_name):
+def log_has_no_errors(log_file_name, print_logfilename_for_problems=False, excluded_substring_list=[]):
     ok=True
     for line in open(log_file_name).readlines():
 
         # First check if the line appears to be in the standard format of messages produced with our logging package
         # For lines produced with our logging package, the first two words in the line are the date and time, then the severity
 
+        bad_line=False
         match_logline_prefix = re.search(r"^20[0-9][0-9]-[A-Z][a-z][a-z]-[0-9]+\s+[0-9:,]+\s+([A-Z]+)", line)
         if match_logline_prefix:
             severity=match_logline_prefix.group(1)
             if severity in ("WARN", "ERROR", "FATAL"):
-                print(line)
-                ok=False
+                bad_line=True
         else: # This line's not produced with our logging package, so let's just look for bad words
             if "WARN" in line or "Warn" in line or "warn" in line or \
                "ERROR" in line or "Error" in line or "error" in line or \
                "FATAL" in line or "Fatal" in line or "fatal" in line or \
                "egmentation fault" in line:
-                print(line)
-                ok=False
-            
+                bad_line=True
+
+        if bad_line:
+            ignore_this_problem=False
+            for excluded_substring in excluded_substring_list:
```

Comment:
Is this the sort of thing we may want the power of regexes for? I think exclusion strings should be as specific as possible, and regular expressions allow for skipping parts of the message which may change (counters, etc), while also allowing certain iterations of the message to go through (exclude if counter is below 100, but not if its over).